lib/onebox/helpers.rb

```diff
@@ -199,18 +199,11 @@ def self.normalize_url_for_output(url)
     end
 
     def self.get_absolute_image_url(src, url)
-      if src && !!(src =~ /^\/\//)
-        uri = URI(url)
-        src = "#{uri.scheme}:#{src}"
-      elsif src && src.match(/^https?:\/\//i).nil?
-        uri = URI(url)
-        src = if !src.start_with?("/") && uri.path.present?
-          "#{uri.scheme}://#{uri.host.sub(/\/$/, '')}#{uri.path.sub(/\/$/, '')}/#{src.sub(/^\//, '')}"
-        else
-          "#{uri.scheme}://#{uri.host.sub(/\/$/, '')}/#{src.sub(/^\//, '')}"
-        end
+      begin
+        URI.parse(url).merge(src).to_s
+      rescue ArgumentError, URI::BadURIError, URI::InvalidURIError => e
```

Comment:
Doesn't look like you're using the `e`?

```suggestion
      rescue ArgumentError, URI::BadURIError, URI::InvalidURIError
```